executor/aggfuncs/aggfunc_test.go

```diff
@@ -263,6 +264,61 @@ func rowMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas
 	return memDeltas, nil
 }
 
+type multiArgsUpdateMemDeltaGens func(*chunk.Chunk, []*types.FieldType) (memDeltas []int64, err error)
+
+func defaultMultiArgsMemDeltaGens(srcChk *chunk.Chunk, dataTypes []*types.FieldType) (memDeltas []int64, err error) {
+	memDeltas = make([]int64, 0)
+	m := make(map[string]bool)
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			memDeltas = append(memDeltas, int64(0))
+			continue
+		}
+		memDelta := int64(0)
+		val := ""
+		memValDelta := int64(0)
+		memKeyDelta := int64(0)
+		for j, dataType := range dataTypes {
+			switch dataType.Tp {
+			case mysql.TypeLonglong:
+				val = strconv.FormatInt(row.GetInt64(j), 10)
+				row.GetRaw(0)
+				memValDelta = aggfuncs.DefUint64Size
+				memKeyDelta = int64(len(val))
+			case mysql.TypeDouble:
+				val = strconv.FormatFloat(row.GetFloat64(j), 'f', 0, 64)
+				memValDelta = aggfuncs.DefFloat64Size
+				memKeyDelta = int64(len(val))
+			case mysql.TypeString:
+				val = row.GetString(j)
+				memValDelta = int64(len(val))
+				memKeyDelta = int64(len(val))
+			case mysql.TypeJSON:
+				json := row.GetJSON(j)
+				memValDelta = int64(unsafe.Sizeof(json))
+				val = json.String()
+				memKeyDelta = int64(len(val))
+			default:
+				return memDeltas, errors.Errorf("unsupported type - %v", dataType.Tp)
+			}
+
+			if j == 0 {
+				if _, ok := m[val]; ok {
+					memDeltas = append(memDeltas, int64(0))
+					break
+				}
+				m[val] = true
+				memDelta += memKeyDelta
+			} else {
+				memDelta += memValDelta
+				memDeltas = append(memDeltas, memDelta)
```

Comment:
value in `partialResult4JsonObjectAgg` is stored as an `interface{}`, we need consider the memory used by `interface{}` itself. the struct of `interface` is as follows, it has two fields, one is the type, the other is a pointer to data, the total memory usage is 16 bytes.
```
type eface struct {
	_type *_type
	data  unsafe.Pointer
}
```

Reply:
So do you mean whatever the value of interface is, the memDelta of it should be 16? What about the string?